src/jit/compiler.h

```diff
@@ -5468,11 +5473,29 @@ class Compiler
         }
     };
 
-#define OMF_HAS_NEWARRAY 0x00000001  // Method contains 'new' of an array
-#define OMF_HAS_NEWOBJ 0x00000002    // Method contains 'new' of an object type.
-#define OMF_HAS_ARRAYREF 0x00000004  // Method contains array element loads or stores.
-#define OMF_HAS_VTABLEREF 0x00000008 // Method contains method table reference.
-#define OMF_HAS_NULLCHECK 0x00000010 // Method contains null check.
+#define OMF_HAS_NEWARRAY 0x00000001   // Method contains 'new' of an array
+#define OMF_HAS_NEWOBJ 0x00000002     // Method contains 'new' of an object type.
+#define OMF_HAS_ARRAYREF 0x00000004   // Method contains array element loads or stores.
+#define OMF_HAS_VTABLEREF 0x00000008  // Method contains method table reference.
+#define OMF_HAS_NULLCHECK 0x00000010  // Method contains null check.
+#define OMF_HAS_FATPOINTER 0x00000020 // Method contains call, that needs fat pointer transformation.
+
+    bool hasFatPointer()
+    {
+        return (optMethodFlags & OMF_HAS_FATPOINTER) != 0;
+    }
+
+    void setHasFatPointer(bool value)
+    {
+        if (value)
+        {
+            optMethodFlags |= OMF_HAS_FATPOINTER;
+        }
+        else
+        {
+            optMethodFlags ^= ~OMF_HAS_FATPOINTER;
+        }
```

Comment:
It might be useful to be more verbose, for example, `SetMethodHasFatPointer()`. I realized reading the code that I wasn't sure what "thing" was being marked, so adding the word "method" would be useful.

---
In reply to: [99244432](https://github.com/dotnet/coreclr/pull/9191#discussion_r99244432) [](ancestors = 99244432)